File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/PublishOntologyProjectDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Details for starting an ontology project publish.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=PublishOntologyProjectDetails.Builder.class)

public final class PublishOntologyProjectDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"workspaceId", "projectName", "comment", "targetConnection"})
    public PublishOntologyProjectDetails(String workspaceId, String projectName, String comment, OntologyPublishTargetConnectionReference targetConnection) {
        super();
        this.workspaceId = workspaceId;
        this.projectName = projectName;
        this.comment = comment;
        this.targetConnection = targetConnection;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Workspace containing the ontology project when publishing by project name.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("workspaceId")
private String workspaceId;

        /**
         * Workspace containing the ontology project when publishing by project name.
         * @param workspaceId the value to set
         * @return this builder
         **/
        

public Builder workspaceId(String workspaceId) {
    this.workspaceId = workspaceId;
    return this;
}
            /**
     * Ontology project key/name to publish.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("projectName")
private String projectName;

        /**
         * Ontology project key/name to publish.
         * @param projectName the value to set
         * @return this builder
         **/
        

public Builder projectName(String projectName) {
    this.projectName = projectName;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("comment")
private String comment;



public Builder comment(String comment) {
    this.comment = comment;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("targetConnection")
private OntologyPublishTargetConnectionReference targetConnection;



public Builder targetConnection(OntologyPublishTargetConnectionReference targetConnection) {
    this.targetConnection = targetConnection;
    return this;
}


        public PublishOntologyProjectDetails build() {
            PublishOntologyProjectDetails model = new PublishOntologyProjectDetails(this.workspaceId
                , this.projectName
                , this.comment
                , this.targetConnection);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(PublishOntologyProjectDetails model) {
                this.workspaceId(model.getWorkspaceId());
    this.projectName(model.getProjectName());
    this.comment(model.getComment());
    this.targetConnection(model.getTargetConnection());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Workspace containing the ontology project when publishing by project name.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("workspaceId")
    private final String workspaceId;

        /**
     * Workspace containing the ontology project when publishing by project name.
     * @return the value
     **/
    
    public String getWorkspaceId() {
        return workspaceId;
    }


        /**
     * Ontology project key/name to publish.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("projectName")
    private final String projectName;

        /**
     * Ontology project key/name to publish.
     * @return the value
     **/
    
    public String getProjectName() {
        return projectName;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("comment")
    private final String comment;

    
    public String getComment() {
        return comment;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("targetConnection")
    private final OntologyPublishTargetConnectionReference targetConnection;

    
    public OntologyPublishTargetConnectionReference getTargetConnection() {
        return targetConnection;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("PublishOntologyProjectDetails(");
        sb.append("workspaceId=").append(String.valueOf(this.workspaceId));
        sb.append(", projectName=").append(String.valueOf(this.projectName));
        sb.append(", comment=").append(String.valueOf(this.comment));
        sb.append(", targetConnection=").append(String.valueOf(this.targetConnection));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof PublishOntologyProjectDetails)) {
            return false;
        }

        PublishOntologyProjectDetails other = (PublishOntologyProjectDetails) o;
        return java.util.Objects.equals(this.workspaceId, other.workspaceId) &&
            java.util.Objects.equals(this.projectName, other.projectName) &&
            java.util.Objects.equals(this.comment, other.comment) &&
            java.util.Objects.equals(this.targetConnection, other.targetConnection);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.workspaceId == null ? 43 : this.workspaceId.hashCode());
        result = (result * PRIME) + (this.projectName == null ? 43 : this.projectName.hashCode());
        result = (result * PRIME) + (this.comment == null ? 43 : this.comment.hashCode());
        result = (result * PRIME) + (this.targetConnection == null ? 43 : this.targetConnection.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/publish_ontology_project_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class PublishOntologyProjectDetails(object):
    """
    Details for starting an ontology project publish.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new PublishOntologyProjectDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param workspace_id:
            The value to assign to the workspace_id property of this PublishOntologyProjectDetails.
        :type workspace_id: str

        :param project_name:
            The value to assign to the project_name property of this PublishOntologyProjectDetails.
        :type project_name: str

        :param comment:
            The value to assign to the comment property of this PublishOntologyProjectDetails.
        :type comment: str

        :param target_connection:
            The value to assign to the target_connection property of this PublishOntologyProjectDetails.
        :type target_connection: oci.aidataplatform_dp.models.OntologyPublishTargetConnectionReference

        """
        self.swagger_types = {
            'workspace_id': 'str',
            'project_name': 'str',
            'comment': 'str',
            'target_connection': 'OntologyPublishTargetConnectionReference'
        }

        self.attribute_map = {
            'workspace_id': 'workspaceId',
            'project_name': 'projectName',
            'comment': 'comment',
            'target_connection': 'targetConnection'
        }

        self._workspace_id = None
        self._project_name = None
        self._comment = None
        self._target_connection = None

    @property
    def workspace_id(self):
        """
        Gets the workspace_id of this PublishOntologyProjectDetails.
        Workspace containing the ontology project when publishing by project name.


        :return: The workspace_id of this PublishOntologyProjectDetails.
        :rtype: str
        """
        return self._workspace_id

    @workspace_id.setter
    def workspace_id(self, workspace_id):
        """
        Sets the workspace_id of this PublishOntologyProjectDetails.
        Workspace containing the ontology project when publishing by project name.


        :param workspace_id: The workspace_id of this PublishOntologyProjectDetails.
        :type: str
        """
        self._workspace_id = workspace_id

    @property
    def project_name(self):
        """
        Gets the project_name of this PublishOntologyProjectDetails.
        Ontology project key/name to publish.


        :return: The project_name of this PublishOntologyProjectDetails.
        :rtype: str
        """
        return self._project_name

    @project_name.setter
    def project_name(self, project_name):
        """
        Sets the project_name of this PublishOntologyProjectDetails.
        Ontology project key/name to publish.


        :param project_name: The project_name of this PublishOntologyProjectDetails.
        :type: str
        """
        self._project_name = project_name

    @property
    def comment(self):
        """
        Gets the comment of this PublishOntologyProjectDetails.

        :return: The comment of this PublishOntologyProjectDetails.
        :rtype: str
        """
        return self._comment

    @comment.setter
    def comment(self, comment):
        """
        Sets the comment of this PublishOntologyProjectDetails.

        :param comment: The comment of this PublishOntologyProjectDetails.
        :type: str
        """
        self._comment = comment

    @property
    def target_connection(self):
        """
        Gets the target_connection of this PublishOntologyProjectDetails.

        :return: The target_connection of this PublishOntologyProjectDetails.
        :rtype: oci.aidataplatform_dp.models.OntologyPublishTargetConnectionReference
        """
        return self._target_connection

    @target_connection.setter
    def target_connection(self, target_connection):
        """
        Sets the target_connection of this PublishOntologyProjectDetails.

        :param target_connection: The target_connection of this PublishOntologyProjectDetails.
        :type: oci.aidataplatform_dp.models.OntologyPublishTargetConnectionReference
        """
        self._target_connection = target_connection

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
